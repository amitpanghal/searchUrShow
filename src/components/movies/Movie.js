import React from "react";

const Movie = React.forwardRef(({ movie, isLastMovie }, ref) => {
  return (
    <>
      <img
        className="img-fluid"
        src={`http://image.tmdb.org/t/p/w154${movie.poster_path}`}
        ref={isLastMovie ? ref : null}
      ></img>
    </>
  );
});
export default Movie;
