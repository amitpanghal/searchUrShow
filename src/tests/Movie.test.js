import React from "react";
import renderer from "react-test-renderer";
import Movie from "../components/movies/Movie";

describe("Movie", () => {
  it("should render an  image of movie poster", () => {
    const tree = renderer.create(
      <Movie
        movie={{ poster_path: "7GsM4mtM0worCtIVeiQt28HieeN.jpg" }}
        isLastMovie={false}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
