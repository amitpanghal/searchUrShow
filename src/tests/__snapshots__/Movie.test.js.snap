// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should render an  image of movie poster 1`] = `
<img
  className="img-fluid"
  src="http://image.tmdb.org/t/p/w1547GsM4mtM0worCtIVeiQt28HieeN.jpg"
/>
`;
